src/Common/src/Interop/Windows/advapi32/Interop.GetServiceDisplayName.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, EntryPoint = "GetServiceDisplayNameW", CharSet = System.Runtime.InteropServices.CharSet.Unicode, SetLastError = true)]
+        private static extern bool GetServiceDisplayNamePrivate(IntPtr SCMHandle, string serviceName, char[] displayName, ref int displayNameLength);
+
+        public static string GetServiceDisplayName(IntPtr SCMHandle, string serviceName)
+        {
+            // Get the size of buffer required
+            int bufLen = 0;
+            bool success = GetServiceDisplayNamePrivate(SCMHandle, serviceName, null, ref bufLen);
```

Comment:
It may look better with `ValueStringBuilder`:

```csharp
var builder = new ValueStringBuilder(4096);
int bufLen;
for (;;)
{
     fixed (char * pBuffer = builder.GetPinnableReference())
     {
         bufLen = builder.Capacity;
         if (GetServiceDisplayNamePrivate(SCMHandle, serviceName, pBuffer, ref int bufLen))
              break;
     }
     int lastError = Marshal.GetLastWin32Error();
     if (lastError != Interop.Errors.ERROR_INSUFFICIENT_BUFFER)
         ... throw exception ...
     builder.EnsureCapacity(bufLen + 1); // Does not include null
}
builder.Length = bufLen;
return builder.ToString();
```
Notice that it is calling `GetServiceDisplayName` just once in the typical case, so it should be more efficient as well.